modules/pipeline/providers/cron/compensator/provider.go

```diff
@@ -394,7 +394,7 @@ func (p *provider) doCronCompensate(compensator apistructs.CronCompensator, notR
 	_, err := p.pipelineFunc.RunPipeline(&apistructs.PipelineRunRequest{
 		PipelineID:   firstOrLastPipeline.ID,
 		Secrets:      firstOrLastPipeline.Extra.IncomingSecrets,
-		IdentityInfo: apistructs.IdentityInfo{InternalClient: firstOrLastPipeline.Extra.InternalClient},
+		IdentityInfo: apistructs.IdentityInfo{InternalClient: firstOrLastPipeline.Extra.InternalClient, UserID: firstOrLastPipeline.NormalLabels[apistructs.LabelUserID]},
```

Comment:
Does `normalLabels` can be nil and cause panic?
I think you should use `spec.Pipeline{}.GetUserID()` or similar method.